odl/space/base_tensors.py

```diff
@@ -0,0 +1,665 @@
+﻿# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Base classes for implementations of tensor spaces."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+from odl.set.sets import Set, RealNumbers, ComplexNumbers
+from odl.set.space import LinearSpace, LinearSpaceElement
+from odl.util.normalize import safe_int_conv
+from odl.util.ufuncs import GeneralizedTensorUfunc
+from odl.util.utility import (
+    is_scalar_dtype, is_floating_dtype, is_real_dtype,
+    arraynd_repr, arraynd_str, dtype_str, repr_signature,
+    TYPE_MAP_R2C, TYPE_MAP_C2R)
+
+
+__all__ = ('TensorSet', 'TensorSpace')
+
+
+class TensorSet(Set):
+
+    """Base class for sets of tensors of arbitrary type."""
+
+    def __init__(self, shape, dtype, order='C'):
```

Comment:
That's right. I guess the reason that Numpy arrays don't have an `order` property is that it's not really a well-defined property. In 1D it's ambiguous and in nD it could be neither nor. But right, let's keep it for later.